### ISSUE TITLE
✅ 🐞 Bug Report: Products Renderer Critical Issue after v3.11.0 in some edge cases - [ED-9837]

### ISSUE THREAD

**Issue by amyriounis** [elementor/elementor#21451]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Inside `elementor-pro/modules/woocommerce/classes/products-renderer.php` this line of code ( the `intval()`part ) :

```php
$query_args['posts_per_page'] = $settings['posts_per_page'] ?? intval( $columns * $rows );
```

will produce a **CRITICAL** `Uncaught TypeError: Unsupported operand types: int * string` error when either of `$rows` or `$columns` variable is set to an empty string value (i.e not explicitly set in the editor by the user).

This is because, the default values for rows and columns will never be assigned properly here:
```php
$this->attributes = $this->parse_attributes( [
	'columns' => $settings['columns'] ?? self::DEFAULT_COLUMNS_AND_ROWS,
	'rows' => $settings['rows'] ?? self::DEFAULT_COLUMNS_AND_ROWS,
	'paginate' => $settings['paginate'],
	'cache' => false,
] );
```

since the empty (`""`) string value is not equal to `null` in PHP and so the null coalescing operator (`??`) will never return `self::DEFAULT_COLUMNS_AND_ROWS`.

This was changed recently, in Elementor Pro v3.11.0. 

It should be reverted **asap** back to how it was before v3.11.0, meaning it should check for `empty()` values instead of `null` ones. Example:
```php
$rows = ! empty( $settings['rows'] ) ? $settings['rows'] : self::DEFAULT_COLUMNS_AND_ROWS;
$columns = ! empty( $settings['columns'] ) ? $settings['columns'] : self::DEFAULT_COLUMNS_AND_ROWS;
```

### Steps to reproduce

1) Add a products widget
2) Have rows or columns option without a specified value.
3) Switch to frontend
4) In the wp logs you should be seeing the specified critical error

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>System Info</summary>

````txt


== Server Environment == 	

Operating System: Linux 	
Software: Apache/2.4.55 (Ubuntu) 	
MySQL version: (Ubuntu) v8.0.32 	
PHP Version: 8.0.28 	
PHP Memory Limit: 512M 	PHP 
Max Input Vars: 1000 	
PHP Max Post Size: 8M 	
GD Installed: No 	
ZIP Installed: Yes 	
Write Permissions: All right 	
Elementor Library: Connected 

== WordPress Environment == 	

Version: 6.1.1 	
Site URL: http://test.demo.localhost 	
Home URL: http://test.demo.localhost 	
WP Multisite: No 	
Max Upload Size: 8 MB 	
Memory limit: 40M 	
Max Memory limit: 512M 	
Permalink Structure: /%postname%/ 	
Language: en-US 	
Timezone: 0 	
Admin Email: test@test.com 	
Debug Mode: Active 

== Theme == 	

Name: Hello Elementor 	
Version: 2.6.1 	
Author: Elementor Team 	
Child Theme: No 

== User == 	

Role: administrator 	
WP Profile lang: en_US 	
User Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36 Edg/110.0.1587.50 

== Active Plugins == 	

Elementor 		
Version: 3.11.2 		
Author: Elementor.com 	

Elementor Pro 		
Version: 3.11.2 		
Author: Elementor.com 	

WooCommerce 		
Version: 7.4.0 		
Author: Automattic 

== Elements Usage == 	 	

footer : 3 		
button : 2 		
divider : 4 		
form : 2 		
heading : 17 		
icon : 3 		
icon-box : 7 		
image : 3 	
nav-menu : 13 		
social-icons : 3 		
spacer : 2 		
text-editor : 8 	
header : 3 		
button : 1 		
heading : 2 		
icon : 11 		
image : 3 		
nav-menu : 7 		
social-icons : 1 		
star-rating : 1 		
theme-site-logo : 3 		
theme-site-title : 1 		
woocommerce-menu-cart : 6 	
page : 9 		
blockquote : 2 		
button : 3 		
heading : 35 		
icon : 2 		
image : 7 		
star-rating : 5 		
text-editor : 20 		
woocommerce-my-account : 1 	
popup : 8 		
form : 1 		
heading : 11 		
icon : 8 		
icon-box : 3 		
icon-list : 1 		
nav-menu : 3 		
spacer : 2 		
tabs : 2 		
text-editor : 2 	
product : 6 		
heading : 11 		
icon : 6 		
icon-box : 15 		
paypal-button : 1 		
text-editor : 1 		
toggle : 6 		
woocommerce-breadcrumb : 6 		
woocommerce-notices : 6 		
woocommerce-product-add-to-cart : 5 		
woocommerce-product-additional-information : 5 		
woocommerce-product-content : 2 		
woocommerce-product-data-tabs : 3 		
woocommerce-product-images : 6 		
woocommerce-product-price : 6 		
woocommerce-product-rating : 1 		
woocommerce-product-related : 5 		
woocommerce-product-title : 6 		
woocommerce-products : 1 	
section : 20 		
button : 7 		
heading : 28 		
image : 14 		
nav-menu : 21 		
post-comments : 1 		
spacer : 2 		
star-rating : 1 		
text-editor : 1 		
woocommerce-products : 2 	
wp-post : 13 		
blockquote : 13 		
heading : 78 		
image : 52 		
text-editor : 65 	
wp-page : 21 		
animated-headline : 4 		
button : 31 		
call-to-action : 7 		
divider : 15 		
form : 3 		
google_maps : 1 		
heading : 406 		
icon : 71 		
icon-box : 42 		
icon-list : 8 		
image : 62 		
menu-anchor : 14 		
posts : 5 		
shortcode : 1 		
spacer : 8 		
star-rating : 1 		
table-of-contents : 1 		
tabs : 1 		
testimonial-carousel : 4 		
text-editor : 52 		
text-path : 3 		
toggle : 6 		
wc-categories : 6 		
wc-elements : 1 		
woocommerce-cart : 1 		
woocommerce-checkout-page : 1 		
woocommerce-my-account : 1 		
woocommerce-notices : 1 		
woocommerce-products : 2 		
woocommerce-purchase-summary : 1 	
single-post : 1 		
heading : 3 		
image : 1 		
post-info : 2 		
posts : 1 		
share-buttons : 1 		
theme-post-content : 1 		
theme-post-title : 1 		
toggle : 1 	
search-results : 1 		
archive-posts : 1 		
heading : 1 		
search-form : 1 		
text-editor : 1 		
theme-archive-title : 1 
product-post : 0 		
divider : 14 		
heading : 23 		
icon : 7 		
icon-box : 26 		
icon-list : 1 		
image : 6 		
text-editor : 4 	
product-archive : 3 	
call-to-action : 2 		
theme-archive-title : 3 	
wc-archive-products : 3 		
woocommerce-archive-description : 1 		
woocommerce-breadcrumb : 3 	
error-404 : 1 		
divider : 1 		
heading : 1 		
text-editor : 1 		
theme-post-title : 1 

== Settings == 	 	

cpt_support: post, page, product 	
allow_tracking: yes 	
unfiltered_files_upload: 1 	
load_fa4_shim: yes

== Features == 	

Custom Fonts: 1 
Custom Icons: 1 

== Integrations == 	 	

woocommerce: Active 

== Elementor Experiments == 	

Optimized DOM Output: Active 	
Improved Asset Loading: Active 	
Improved CSS Loading: Inactive 	
Inline Font Icons: Inactive 	
Accessibility Improvements: Active 	
Additional Custom Breakpoints: Inactive 	
admin_menu_rearrangement: Inactive by default 	
Flexbox Container: Inactive 	
Upgrade Swiper Library: Inactive by default 	
Default to New Theme Builder: Active 	
Hello Theme Header & Footer: Inactive 	
Editor V2: Inactive by default 	
Landing Pages: Active 	
Nested Elements: Inactive by default 	
Lazy Load Background Images: Inactive by default 	
Save as Default: Active by default 	
Page Transitions: Inactive 	
Notes: Active 	
Loop: Active by default 	
Form Submissions: Active 	
Scroll Snap: Active 	
Mega Menu: Inactive by default 

== Log == 	 

Log: showing 20 of 30

2023-02-14 14:23:58 [info] Elementor/Upgrades - _on_each_version Start 

2023-02-14 14:24:02 [info] Elementor/Upgrades - _on_each_version Finished 

2023-02-14 14:24:02 [info] Elementor data updater process has been completed. [array ( 'plugin' => 'Elementor', 'from' => '3.10.1', 'to' => '3.11.0', )] 

2023-02-14 14:24:49 [info] Elementor data updater process has been queued. [array ( 'plugin' => 'Elementor Pro', 'from' => '3.10.2', 'to' => '3.11.0', )] 

2023-02-14 14:24:49 [info] elementor-pro::elementor_pro_updater Started 

2023-02-14 14:24:49 [info] Elementor Pro/Upgrades - _on_each_version Start 

2023-02-14 14:24:49 [info] Elementor Pro/Upgrades - _on_each_version Finished 

2023-02-14 14:24:49 [info] Elementor data updater process has been completed. [array ( 'plugin' => 'Elementor Pro', 'from' => '3.10.2', 'to' => '3.11.0', )] 

2023-02-17 12:32:45 [info] Elementor data updater process has been queued. [array ( 'plugin' => 'Elementor', 'from' => '3.11.0', 'to' => '3.11.1', )] 

2023-02-17 12:32:47 [info] elementor::elementor_updater Started 

2023-02-17 12:32:47 [info] Elementor/Upgrades - _on_each_version Start 

2023-02-17 12:32:56 [info] Elementor data updater process has been queued. [array ( 'plugin' => 'Elementor Pro', 'from' => '3.11.0', 'to' => '3.11.1', )] 

2023-02-17 12:32:57 [info] Elementor/Upgrades - _on_each_version Finished 

2023-02-17 12:32:57 [info] Elementor data updater process has been completed. [array ( 'plugin' => 'Elementor', 'from' => '3.11.0', 'to' => '3.11.1', )] 

2023-02-17 12:32:59 [info] Elementor data updater process has been queued. [array ( 'plugin' => 'Elementor', 'from' => '3.11.0', 'to' => '3.11.1', )] 

2023-02-17 12:33:00 [info] elementor-pro::elementor_pro_updater Started 

2023-02-17 12:33:00 [info] Elementor Pro/Upgrades - _on_each_version Start 

2023-02-17 12:33:00 [info] Elementor Pro/Upgrades - _on_each_version Finished 

2023-02-17 12:33:00 [info] Elementor data updater process has been completed. [array ( 'plugin' => 'Elementor Pro', 'from' => '3.11.0', 'to' => '3.11.1', )] 

2023-02-17 12:33:01 [info] Elementor data updater process has been queued. [array ( 'plugin' => 'Elementor Pro', 'from' => '3.11.0', 'to' => '3.11.1', )] 

PHP: showing 12 of 12

PHP: 2023-01-19 06:45:49 [error X 1][/home/test/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::182] Uncaught Error: Class &quot;\ElementorPro\Modules\ThemeBuilder\Conditions\General&quot; not found in /home/test/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:182 Stack trace: #0 /home/test/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(264): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager-&gt;register_condition() #1 /home/test/wp-includes/class-wp-hook.php(308): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager-&gt;register_conditions() #2 /home/test/wp-includes/class-wp-hook.php(332): WP_Hook-&gt;apply_filters() #3 /home/test/wp-includes/plugin.php(517): WP_Hook-&gt;do_action() #4 /home/test/wp-settings.php(639): do_action() #5 /home/test/wp-config.php(79): require_once(&#039;...&#039;) #6 /home/test/wp-load.php(50): require_once(&#039;...&#039;) #7 /home/test/wp-admin/admin-ajax.php(22): require_once(&#039;...&#039;) #8 {main} thrown [array ( 'trace' => ' #0: Elementor\Core\Logger\Manager -> shutdown() ', )] 

PHP: 2023-01-19 15:15:34 [warning X 3][/home/test/wp-content/plugins/elementor/core/common/modules/ajax/module.php::172] Undefined array key &quot;data&quot; [array ( 'trace' => ' #0: Elementor\Core\Logger\Manager -> shutdown() ', )] 

PHP: 2023-01-22 13:53:14 [warning X 2][/home/test/wp-content/plugins/elementor/includes/managers/image.php::108] Trying to access array offset on value of type bool [array ( 'trace' => ' #0: Elementor\Core\Logger\Manager -> shutdown() ', )] 

PHP: 2023-01-30 15:46:21 [warning X 1][/home/test/wp-content/plugins/elementor-pro/modules/assets-manager/asset-types/icons/custom-icons.php::461] Undefined array key &quot;prefix&quot; [array ( 'trace' => ' #0: Elementor\Core\Logger\Manager -> shutdown() ', )] 

PHP: 2023-02-20 13:52:32 [error X 1][/home/test/wp-content/plugins/elementor-pro/modules/woocommerce/classes/products-renderer.php::83] Uncaught TypeError: Unsupported operand types: int * string in /home/test/wp-content/plugins/elementor-pro/modules/woocommerce/classes/products-renderer.php:83 Stack trace: #0 /home/test/wp-content/plugins/elementor-pro/modules/woocommerce/classes/products-renderer.php(27): ElementorPro\Modules\Woocommerce\Classes\Products_Renderer-&gt;parse_query_args() #1 /home/test/wp-content/plugins/elementor-pro/modules/woocommerce/widgets/products.php(330): ElementorPro\Modules\Woocommerce\Classes\Products_Renderer-&gt;__construct() #2 /home/test/wp-content/plugins/elementor-pro/modules/woocommerce/widgets/products.php(359): ElementorPro\Modules\Woocommerce\Widgets\Products::get_shortcode_object() #3 /home/test/wp-content/plugins/elementor/includes/base/controls-stack.php(2268): ElementorPro\Modules\Woocommerce\Widgets\Products-&gt;render() #4 /home/test/wp-content/plugins/elementor/includes/base/widget-base.php(609): Elementor\Controls_Stack-&gt;render_by_mode() #5 /home/test/wp-content/plugins/elementor/includes/base/widget-base.php(750): Elementor\Widget_Base-&gt;render_content() #6 /home/test/wp-content/plugins/elementor/includes/base/element-base.php(449): Elementor\Widget_Base-&gt;print_content() #7 /home/test/wp-content/plugins/elementor/includes/base/element-base.php(1354): Elementor\Element_Base-&gt;print_element() #8 /home/test/wp-content/plugins/elementor/includes/base/element-base.php(449): Elementor\Element_Base-&gt;print_content() #9 /home/test/wp-content/plugins/elementor/includes/base/element-base.php(1354): Elementor\Element_Base-&gt;print_element() #10 /home/test/wp-content/plugins/elementor/includes/base/element-base.php(449): Elementor\Element_Base-&gt;print_content() #11 /home/test/wp-content/plugins/elementor/core/base/document.php(1669): Elementor\Element_Base-&gt;print_element() #12 /home/test/wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php(418): Elementor\Core\Base\Document-&gt;print_elements() #13 /home/test/wp-content/plugins/elementor/includes/frontend.php(1179): ElementorPro\Modules\ThemeBuilder\Documents\Theme_Document-&gt;print_elements_with_wrapper() #14 /home/test/wp-content/plugins/elementor/includes/frontend.php(1247): Elementor\Frontend-&gt;get_builder_content() #15 /home/test/wp-content/plugins/elementor-pro/modules/library/classes/shortcode.php(46): Elementor\Frontend-&gt;get_builder_content_for_display() #16 /home/test/wp-includes/shortcodes.php(355): ElementorPro\Modules\Library\Classes\Shortcode-&gt;shortcode() #17 [internal function]: do_shortcode_tag() #18 /home/test/wp-includes/shortcodes.php(227): preg_replace_callback() #19 /home/test/wp-content/plugins/elementor/includes/base/widget-base.php(426): do_shortcode() #20 /home/test/wp-content/plugins/elementor/includes/base/widget-base.php(447): Elementor\Widget_Base-&gt;parse_text_editor() #21 /home/test/wp-content/plugins/elementor/includes/widgets/tabs.php(537): Elementor\Widget_Base-&gt;print_text_editor() #22 /home/test/wp-content/plugins/elementor/includes/base/controls-stack.php(2268): Elementor\Widget_Tabs-&gt;render() #23 /home/test/wp-content/plugins/elementor/includes/base/widget-base.php(609): Elementor\Controls_Stack-&gt;render_by_mode() #24 /home/test/wp-content/plugins/elementor/includes/base/widget-base.php(733): Elementor\Widget_Base-&gt;render_content() #25 /home/test/wp-content/plugins/elementor/includes/base/element-base.php(536): Elementor\Widget_Base-&gt;get_raw_data() #26 /home/test/wp-content/plugins/elementor/includes/base/element-base.php(536): Elementor\Element_Base-&gt;get_raw_data() #27 /home/test/wp-content/plugins/elementor/core/base/document.php(966): Elementor\Element_Base-&gt;get_raw_data() #28 /home/test/wp-content/plugins/elementor/core/base/document.php(632): Elementor\Core\Base\Document-&gt;get_elements_raw_data() #29 /home/test/wp-content/plugins/elementor/includes/base/controls-stack.php(1021): Elementor\Core\Base\Document-&gt;get_initial_config() #30 /home/test/wp-content/plugins/elementor/core/editor/editor.php(383): Elementor\Controls_Stack-&gt;get_config() #31 /home/test/wp-includes/class-wp-hook.php(308): Elementor\Core\Editor\Editor-&gt;enqueue_scripts() #32 /home/test/wp-includes/class-wp-hook.php(332): WP_Hook-&gt;apply_filters() #33 /home/test/wp-includes/plugin.php(517): WP_Hook-&gt;do_action() #34 /home/test/wp-includes/script-loader.php(2180): do_action() #35 /home/test/wp-includes/class-wp-hook.php(308): wp_enqueue_scripts() #36 /home/test/wp-includes/class-wp-hook.php(332): WP_Hook-&gt;apply_filters() #37 /home/test/wp-includes/plugin.php(517): WP_Hook-&gt;do_action() #38 /home/test/wp-includes/general-template.php(3043): do_action() #39 /home/test/wp-content/plugins/elementor/includes/editor-templates/editor-wrapper.php(31): wp_head() #40 /home/test/wp-content/plugins/elementor/core/editor/editor-loader.php(113): include(&#039;...&#039;) #41 /home/test/wp-content/plugins/elementor/core/editor/editor.php(172): Elementor\Core\Editor\Editor_Loader-&gt;print_root_template() #42 /home/test/wp-includes/class-wp-hook.php(308): Elementor\Core\Editor\Editor-&gt;init() #43 /home/test/wp-includes/class-wp-hook.php(332): WP_Hook-&gt;apply_filters() #44 /home/test/wp-includes/plugin.php(517): WP_Hook-&gt;do_action() #45 /home/test/wp-admin/admin.php(419): do_action() #46 /home/test/wp-admin/post.php(12): require_once(&#039;...&#039;) #47 {main} thrown [array ( 'trace' => ' #0: Elementor\Core\Logger\Manager -> shutdown() ', )] 

PHP: 2023-02-23 14:48:31 [error X 18][/test/wp-content/plugins/elementor-pro/modules/woocommerce/classes/products-renderer.php::83] Uncaught TypeError: Unsupported operand types: int * string in /test/wp-content/plugins/elementor-pro/modules/woocommerce/classes/products-renderer.php:83 Stack trace: #0 /test/wp-content/plugins/elementor-pro/modules/woocommerce/classes/products-renderer.php(27): ElementorPro\Modules\Woocommerce\Classes\Products_Renderer-&gt;parse_query_args() #1 /test/wp-content/plugins/elementor-pro/modules/woocommerce/widgets/products.php(330): ElementorPro\Modules\Woocommerce\Classes\Products_Renderer-&gt;__construct() #2 /test/wp-content/plugins/elementor-pro/modules/woocommerce/widgets/products.php(359): ElementorPro\Modules\Woocommerce\Widgets\Products::get_shortcode_object() #3 /test/wp-content/plugins/elementor/includes/base/controls-stack.php(2268): ElementorPro\Modules\Woocommerce\Widgets\Products-&gt;render() #4 /test/wp-content/plugins/elementor/includes/base/widget-base.php(609): Elementor\Controls_Stack-&gt;render_by_mode() #5 /test/wp-content/plugins/elementor/includes/base/widget-base.php(750): Elementor\Widget_Base-&gt;render_content() #6 /test/wp-content/plugins/elementor/includes/base/element-base.php(449): Elementor\Widget_Base-&gt;print_content() #7 /test/wp-content/plugins/elementor/includes/base/element-base.php(1354): Elementor\Element_Base-&gt;print_element() #8 /test/wp-content/plugins/elementor/includes/base/element-base.php(449): Elementor\Element_Base-&gt;print_content() #9 /test/wp-content/plugins/elementor/includes/base/element-base.php(1354): Elementor\Element_Base-&gt;print_element() #10 /test/wp-content/plugins/elementor/includes/base/element-base.php(449): Elementor\Element_Base-&gt;print_content() #11 /test/wp-content/plugins/elementor/core/base/document.php(1669): Elementor\Element_Base-&gt;print_element() #12 /test/wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php(418): Elementor\Core\Base\Document-&gt;print_elements() #13 /test/wp-content/plugins/elementor/includes/frontend.php(1179): ElementorPro\Modules\ThemeBuilder\Documents\Theme_Document-&gt;print_elements_with_wrapper() #14 /test/wp-content/plugins/elementor/includes/frontend.php(1247): Elementor\Frontend-&gt;get_builder_content() #15 /test/wp-content/plugins/elementor-pro/modules/library/classes/shortcode.php(46): Elementor\Frontend-&gt;get_builder_content_for_display() #16 /test/wp-includes/shortcodes.php(355): ElementorPro\Modules\Library\Classes\Shortcode-&gt;shortcode() #17 [internal function]: do_shortcode_tag() #18 /test/wp-includes/shortcodes.php(227): preg_replace_callback() #19 /test/wp-content/plugins/elementor/includes/base/widget-base.php(426): do_shortcode() #20 /test/wp-content/plugins/elementor/includes/base/widget-base.php(447): Elementor\Widget_Base-&gt;parse_text_editor() #21 /test/wp-content/plugins/elementor/includes/widgets/tabs.php(537): Elementor\Widget_Base-&gt;print_text_editor() #22 /test/wp-content/plugins/elementor/includes/base/controls-stack.php(2268): Elementor\Widget_Tabs-&gt;render() #23 /test/wp-content/plugins/elementor/includes/base/widget-base.php(609): Elementor\Controls_Stack-&gt;render_by_mode() #24 /test/wp-content/plugins/elementor/includes/base/widget-base.php(733): Elementor\Widget_Base-&gt;render_content() #25 /test/wp-content/plugins/elementor/includes/base/element-base.php(536): Elementor\Widget_Base-&gt;get_raw_data() #26 /test/wp-content/plugins/elementor/includes/base/element-base.php(536): Elementor\Element_Base-&gt;get_raw_data() #27 /test/wp-content/plugins/elementor/core/base/document.php(966): Elementor\Element_Base-&gt;get_raw_data() #28 /test/wp-content/plugins/elementor/core/base/document.php(632): Elementor\Core\Base\Document-&gt;get_elements_raw_data() #29 /test/wp-content/plugins/elementor/includes/base/controls-stack.php(1021): Elementor\Core\Base\Document-&gt;get_initial_config() #30 /test/wp-content/plugins/elementor/core/editor/editor.php(383): Elementor\Controls_Stack-&gt;get_config() #31 /test/wp-includes/class-wp-hook.php(308): Elementor\Core\Editor\Editor-&gt;enqueue_scripts() #32 /test/wp-includes/class-wp-hook.php(332): WP_Hook-&gt;apply_filters() #33 /test/wp-includes/plugin.php(517): WP_Hook-&gt;do_action() #34 /test/wp-includes/script-loader.php(2180): do_action() #35 /test/wp-includes/class-wp-hook.php(308): wp_enqueue_scripts() #36 /test/wp-includes/class-wp-hook.php(332): WP_Hook-&gt;apply_filters() #37 /test/wp-includes/plugin.php(517): WP_Hook-&gt;do_action() #38 /test/wp-includes/general-template.php(3043): do_action() #39 /test/wp-content/plugins/elementor/includes/editor-templates/editor-wrapper.php(31): wp_head() #40 /test/wp-content/plugins/elementor/core/editor/editor-loader.php(113): include(&#039;...&#039;) #41 /test/wp-content/plugins/elementor/core/editor/editor.php(172): Elementor\Core\Editor\Editor_Loader-&gt;print_root_template() #42 /test/wp-includes/class-wp-hook.php(308): Elementor\Core\Editor\Editor-&gt;init() #43 /test/wp-includes/class-wp-hook.php(332): WP_Hook-&gt;apply_filters() #44 /test/wp-includes/plugin.php(517): WP_Hook-&gt;do_action() #45 /test/wp-admin/admin.php(419): do_action() #46 /test/wp-admin/post.php(12): require_once(&#039;...&#039;) #47 {main} thrown [array ( 'trace' => ' #0: Elementor\Core\Logger\Manager -> shutdown() ', )] 

PHP: 2023-02-23 15:18:46 [error X 2][/test/wp-content/plugins/elementor/core/files/file-types/svg.php::234] Uncaught Error: Call to undefined function Elementor\Core\Files\File_Types\simplexml_load_file() in /test/wp-content/plugins/elementor/core/files/file-types/svg.php:234 Stack trace: #0 /test/wp-includes/class-wp-hook.php(308): Elementor\Core\Files\File_Types\Svg-&gt;set_svg_meta_data() #1 /test/wp-includes/plugin.php(205): WP_Hook-&gt;apply_filters() #2 /test/wp-includes/post.php(6630): apply_filters() #3 /test/wp-content/plugins/woocommerce/includes/class-wc-regenerate-images-request.php(157): wp_update_attachment_metadata() #4 /test/wp-content/plugins/woocommerce/includes/abstracts/class-wc-background-process.php(102): WC_Regenerate_Images_Request-&gt;task() #5 /test/wp-content/plugins/woocommerce/includes/libraries/wp-background-process.php(175): WC_Background_Process-&gt;handle() #6 /test/wp-includes/class-wp-hook.php(308): WP_Background_Process-&gt;maybe_handle() #7 /test/wp-includes/class-wp-hook.php(332): WP_Hook-&gt;apply_filters() #8 /test/wp-includes/plugin.php(517): WP_Hook-&gt;do_action() #9 /test/wp-admin/admin-ajax.php(203): do_action() #10 {main} thrown [array ( 'trace' => ' #0: Elementor\Core\Logger\Manager -> shutdown() ', )] 

PHP: 2023-02-23 15:20:51 [error X 1][/test/wp-content/plugins/elementor/core/files/file-types/svg.php::234] Uncaught Error: Call to undefined function Elementor\Core\Files\File_Types\simplexml_load_file() in /test/wp-content/plugins/elementor/core/files/file-types/svg.php:234 Stack trace: #0 /test/wp-includes/class-wp-hook.php(308): Elementor\Core\Files\File_Types\Svg-&gt;set_svg_meta_data() #1 /test/wp-includes/plugin.php(205): WP_Hook-&gt;apply_filters() #2 /test/wp-includes/post.php(6630): apply_filters() #3 /test/wp-content/plugins/woocommerce/includes/class-wc-regenerate-images-request.php(157): wp_update_attachment_metadata() #4 /test/wp-content/plugins/woocommerce/includes/abstracts/class-wc-background-process.php(102): WC_Regenerate_Images_Request-&gt;task() #5 /test/wp-content/plugins/woocommerce/includes/libraries/wp-background-process.php(175): WC_Background_Process-&gt;handle() #6 /test/wp-includes/class-wp-hook.php(308): WP_Background_Process-&gt;maybe_handle() #7 /test/wp-includes/class-wp-hook.php(332): WP_Hook-&gt;apply_filters() #8 /test/wp-includes/plugin.php(517): WP_Hook-&gt;do_action() #9 /test/wp-admin/admin-ajax.php(188): do_action() #10 {main} thrown [array ( 'trace' => ' #0: Elementor\Core\Logger\Manager -> shutdown() ', )] 

PHP: 2023-02-23 15:27:23 [error X 19][/test/wp-content/plugins/elementor/core/files/file-types/svg.php::234] Uncaught Error: Call to undefined function Elementor\Core\Files\File_Types\simplexml_load_file() in /test/wp-content/plugins/elementor/core/files/file-types/svg.php:234 Stack trace: #0 /test/wp-includes/class-wp-hook.php(308): Elementor\Core\Files\File_Types\Svg-&gt;set_svg_meta_data() #1 /test/wp-includes/plugin.php(205): WP_Hook-&gt;apply_filters() #2 /test/wp-includes/post.php(6630): apply_filters() #3 /test/wp-content/plugins/woocommerce/includes/class-wc-regenerate-images-request.php(157): wp_update_attachment_metadata() #4 /test/wp-content/plugins/woocommerce/includes/abstracts/class-wc-background-process.php(102): WC_Regenerate_Images_Request-&gt;task() #5 /test/wp-content/plugins/woocommerce/includes/libraries/wp-background-process.php(447): WC_Background_Process-&gt;handle() #6 /test/wp-includes/class-wp-hook.php(308): WP_Background_Process-&gt;handle_cron_healthcheck() #7 /test/wp-includes/class-wp-hook.php(332): WP_Hook-&gt;apply_filters() #8 /test/wp-includes/plugin.php(565): WP_Hook-&gt;do_action() #9 /test/wp-cron.php(188): do_action_ref_array() #10 {main} thrown [array ( 'trace' => ' #0: Elementor\Core\Logger\Manager -> shutdown() ', )] 

== Elementor - Compatibility Tag == 	 	

Elementor Pro: Compatible 

== Elementor Pro - Compatibility Tag == 	 

````
</details>

**Comment by nicholaszein** [comment]
Hey @amyriounis, 

Thank you for submitting your report! 🙏 

### ❌ While reviewing your report, we noticed it was missing some key pieces of information:

- 💾 **System info:** Please include the System Info report as outlined in our [guidelines](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=bug.yml). 
   - Get your System Info from your WordPress `Dashboard` > `Elementor` > `System info` [learn how to copy it here](https://elementor.com/help/hare-my-system-info/). We can't accept issues without it.



Without them, we simply can't validate your issue. 😕 

### 🧐 Please provide the missing information above so we can troubleshoot your issue faster and better.

Kind regards

**Comment by davidLTC** [comment]
Thank you so much @amyriounis ! We just ran into this on our website and the suggested fix in your detailed issue log worked perfectly!  

I went into the Plugin File Editor > Elementor Pro > modules/woocommerce/classes/products-renderer.php and revised that line of code:

`		$this->attributes = $this->parse_attributes( [
			'columns' => ! empty( $settings['columns'] ) ? $settings['columns'] : self::DEFAULT_COLUMNS_AND_ROWS,
			'rows' => ! empty( $settings['rows'] ) ? $settings['rows'] : self::DEFAULT_COLUMNS_AND_ROWS,
			'paginate' => $settings['paginate'],
			'cache' => false,
		] );
`
... and I stopped getting server 500 errors. Thank you so much for this fix!

I hope this gets resolved in the next update!

**Comment by amyriounis** [comment]
Hi @davidLTC, glad you got this sorted out, ur welcome:)

<hr/>

Hi @nicholaszein, 

> While reviewing your report, we noticed it was missing some key pieces of information:

I think I already have provided all the relevant information needed to reproduce and fix the issue but [here](https://github.com/elementor/elementor/files/10828222/system-info.txt). Hopefully you can get this sorted out fast, as it basically affects all sites that have any type of product widgets displayed.

**Comment by nicholaszein** [comment]
Hi @amyriounis,

Thank you for providing the missing information. The System Info is THE most important part of any Bug Report, and our team needs it to make sure they are correctly reproducing the issue. Always provide the System Info from Elementor.

Thanks

**Comment by veniko** [comment]
@nicholaszein  Hi there, this doesn't seem to be an edge case. Several client already reported their site crached after an update and the reason was the same - this issue outlined here. Please release a fix as soon as possible.

**Comment by nicholaszein** [comment]
Hi @veniko,

Thanks for the feedback. Edge cases means that it's not happen consistently in all installations. I couldn't reproduce the issue on my end for example.

The team is investigating this issue and we will update you once a fix is released.

I understand the frustration and eagerness, and I kindly ask you to be patient.

**Comment by amyriounis** [comment]
Hi @nicholaszein 

>I couldn't reproduce the issue on my end for example.

If I can help a bit more regarding testing, I think only the "Products" widget lets you assign the "Rows" field from the editor:

![roes](https://user-images.githubusercontent.com/9055437/221561856-dfc7d35b-b42e-413c-93a7-69ab4dee65de.png)

So if you create a page, and add a products widget with the "Rows" field left empty, you should get something like this:

![kluh](https://user-images.githubusercontent.com/9055437/221562050-50c12f78-c360-4c5b-b452-bad8f47d699c.png)

The "Columns" field is not an issue as it is set to `required` in the backend, so absence of value is not possible.

**Comment by amwebch** [comment]
Hi,
I just ran into the same issue: having an empty value for "Rows" in the products widget, which, after the latest update, was throwing the fatal error "Uncaught TypeError: Unsupported operand types: int * string in .../wp-content/plugins/elementor-pro/modules/woocommerce/classes/products-renderer.php:83".

As a temporary workaround, I added "floatval()" around the values for columns and rows in line 83, "solving" the issue.

Kind regards,
Aron

**Comment by amyriounis** [comment]
Hi @amwebch, just bear in mind that `floatval()` will return `0` for empty values while the widget's default (`DEFAULT_COLUMNS_AND_ROWS`) is set to `4`.

**Comment by amwebch** [comment]
Hi @amyriounis,
Thank you for the additional info. I will correct it to the more extended and correct version using `DEFAULT_COLUMNS_AND_ROWS` provided by you.
In the meantime, let's also hope that the Elementor team will fix this in an upcoming release. ;-)

**Comment by cashewcoding** [comment]
Did anything happen in this matter?

The issue has been reported more than two months ago and still in the latest version 3.12.3 this issue prevails...

It can be easily fixed by replacing the code on lines 21 ff. of products-renderer.php with the code provided by davidLTC on February 23. So, please Elementor team, fix this!

I believe this happens when you have products being displayed on a page by Elementor that are later deleted or set to private. There might be other ways to reproduce this though...

I hope this will be fixed asap!

Best regards,
David

**Comment by veniko** [comment]
@cashewcoding  @amwebch Unfortunately, bug fixes often take too long to get included.

**Comment by nicholaszein** [comment]
Hello everyone!

We just released Elementor and Elementor Pro version 3.13-beta4 with a fix for this issue! 🙌 

Please, if possible, test the new version so to see if the issue is fixed for you. The final release is estimated to be released on `May 9, 2023` (next week).

Kind regards
